### PR TITLE
[clang] Predefined macros for float128 support

### PR DESCRIPTION
**Pull request by pranavk**
Expose some predefined macros for Float128 support.

### CODE REVIEW

File: clang/lib/Frontend/InitPreprocessor.cpp [hunk 1]
```diff
@@ -1076,6 +1076,8 @@ static void InitializePredefinedMacros(const TargetInfo &TI,
   DefineFloatMacros(Builder, "FLT", &TI.getFloatFormat(), "F");
   DefineFloatMacros(Builder, "DBL", &TI.getDoubleFormat(), "");
   DefineFloatMacros(Builder, "LDBL", &TI.getLongDoubleFormat(), "L");
+  if (TI.hasFloat128Type())
+    DefineFloatMacros(Builder, "FLT128", &TI.getFloat128Format(), "Q");
```
Comment (alexander-shaposhnikov) [inline clang/lib/Frontend/InitPreprocessor.cpp:1080]: This needs tests (see e.g. https://reviews.llvm.org/D111382, https://github.com/llvm/llvm-project/blob/main/clang/test/Preprocessor)

Comment (pranavk) [reply clang/lib/Frontend/InitPreprocessor.cpp:1080]: I am not adding any new float type or any literal suffix (unlike other revision you pointed by Ray). This is only adding float macros for which I modified the test that was expecting it.

**Review by echristo — COMMENTED** [review]
In general this looks good. Since we don't have full compiler-rt support for this yet we may want to wait until that to merge. Alex is already looking at this and coordinating with his needs is probably the best solution.

**Review by alexander-shaposhnikov — APPROVED** [review]
LG

**Pull request merged**